src/compiler/utilities.ts

```diff
@@ -1434,24 +1434,10 @@ namespace ts {
     }
 
     function getJSDocTags(node: Node, kind: SyntaxKind): JSDocTag[] {
-        const docs = getJSDocs(node);
-        if (docs) {
-            const result: JSDocTag[] = [];
-            for (const doc of docs) {
-                if (doc.kind === SyntaxKind.JSDocParameterTag) {
-                    if (doc.kind === kind) {
-                        result.push(doc as JSDocTag);
-                    }
-                }
-                else {
-                    const tags = (doc as JSDoc).tags;
-                    if (tags) {
-                        result.push(...filter(tags, tag => tag.kind === kind));
-                    }
-                }
-            }
-            return result;
-        }
+        return flatMap(getJSDocs(node), doc =>

```

Comment:
This isn't exactly identical to before since we now check for `SyntaxKind.JSDocComment` instead of `SyntaxKind.JSDocParameterTag`. I think it makes more sense this way though, since the cast `doc as JSDoc` isn't valid if all we know is that it's not an `@param` tag. @sandersn